### ISSUE TITLE
MDTimePicker bug

### ISSUE THREAD

**Issue by adywizard** [kivymd/KivyMD#966]
### Description of the Bug

Hi, On latest version of MDTimePicker if I try to delete minutes app crashes with: RecursionError,  also colors doesn't update when primary_color is changed
as seen from the below video. 

https://www.youtube.com/watch?v=KW0UZBCuQEU

* OS:  Android
* Python:  3.8.5
* Kivy: 2.0.0
* KivyMD: master

````04-19 19:23:08.882 20200 20423 I python  :    File "kivy/properties.pyx", line 721, in kivy.properties.ObservableList.__setitem__
04-19 19:23:08.882 20200 20423 I python  :    File "kivy/properties.pyx", line 707, in kivy.properties.observable_list_dispatch
04-19 19:23:08.882 20200 20423 I python  :    File "kivy/properties.pyx", line 600, in kivy.properties.Property.dispatch
04-19 19:23:08.882 20200 20423 I python  :    File "kivy/_event.pyx", line 1248, in kivy._event.EventObservers.dispatch
04-19 19:23:08.882 20200 20423 I python  :    File "kivy/_event.pyx", line 1154, in kivy._event.EventObservers._dispatch
04-19 19:23:08.882 20200 20423 I python  :    File "kivy/properties.pyx", line 1504, in kivy.properties.AliasProperty.trigger_change
04-19 19:23:08.882 20200 20423 I python  :    File "kivy/properties.pyx", line 1511, in kivy.properties.AliasProperty.trigger_change
04-19 19:23:08.882 20200 20423 I python  :    File "kivy/properties.pyx", line 1538, in kivy.properties.AliasProperty.dispatch
04-19 19:23:08.882 20200 20423 I python  :    File "kivy/_event.pyx", line 1248, in kivy._event.EventObservers.dispatch
04-19 19:23:08.882 20200 20423 I python  :    File "kivy/_event.pyx", line 1154, in kivy._event.EventObservers._dispatch
04-19 19:23:08.882 20200 20423 I python  :    File "/home/ady/github_projects/car-locator/.buildozer/android/platform/build-armeabi-v7a/build/python-installs/locator/kivymd/uix/picker.py", line 2414, in on_text
04-19 19:23:08.882 20200 20423 I python  :    File "/home/ady/github_projects/car-locator/.buildozer/android/platform/build-armeabi-v7a/build/python-installs/locator/kivy/uix/textinput.py", line 1925, in _refresh_text
04-19 19:23:08.882 20200 20423 I python  :    File "kivy/properties.pyx", line 498, in kivy.properties.Property.__set__
04-19 19:23:08.883 20200 20423 I python  :    File "kivy/properties.pyx", line 1527, in kivy.properties.AliasProperty.set
04-19 19:23:08.883 20200 20423 I python  :    File "/home/ady/github_projects/car-locator/.buildozer/android/platform/build-armeabi-v7a/build/python-installs/locator/kivy/uix/textinput.py", line 2762, in _set_cursor
04-19 19:23:08.883 20200 20423 I python  :    File "/home/ady/github_projects/car-locator/.buildozer/android/platform/build-armeabi-v7a/build/python-installs/locator/kivy/uix/textinput.py", line 619, in cursor_offset
04-19 19:23:08.883 20200 20423 I python  :    File "/home/ady/github_projects/car-locator/.buildozer/android/platform/build-armeabi-v7a/build/python-installs/locator/kivy/uix/textinput.py", line 1798, in _get_text_width
04-19 19:23:08.883 20200 20423 I python  :    File "/home/ady/github_projects/car-locator/.buildozer/android/platform/build-armeabi-v7a/build/python-installs/locator/kivy/core/text/text_sdl2.py", line 35, in get_extents
04-19 19:23:08.883 20200 20423 I python  :    File "kivy/core/text/_text_sdl2.pyx", line 203, in kivy.core.text._text_sdl2._get_extents
04-19 19:23:08.883 20200 20423 I python  :    File "kivy/core/text/_text_sdl2.pyx", line 157, in kivy.core.text._text_sdl2._get_font
04-19 19:23:08.883 20200 20423 I python  :    File "/home/ady/github_projects/car-locator/.buildozer/android/platform/build-armeabi-v7a/build/python-installs/locator/kivy/core/text/text_sdl2.py", line 25, in _get_font_id
04-19 19:23:08.883 20200 20423 I python  :    File "/home/ady/github_projects/car-locator/.buildozer/android/platform/build-armeabi-v7a/build/python-installs/locator/kivy/core/text/text_sdl2.py", line 25, in <listcomp>
04-19 19:23:08.883 20200 20423 I python  :  RecursionError: maximum recursion depth exceeded while getting the str of an object
04-19 19:23:08.883 20200 20423 I python  : Python for android ended.```

**Comment by podraco** [comment]
First. I want to tell you that I loved you UI, it looks really good.
Second of all, we might need to find what exactly happened. have you been able to reproduce the same error on pc?

**Comment by adywizard** [comment]
Sorry for late response, actually now when you say it, I didn't even try, I just got crash on Android, then I recorded everything, I'm not at home right now so I can't try it, eventually when I get home I'll try it on desktop then I'll post it back. By the way thanks for the compliment.

**Comment by podraco** [comment]
it looks really good, is it a video or a gift?
the background?

**Comment by podraco** [comment]
closed by accident, sorry.

**Comment by adywizard** [comment]
You can find it at Kivymd wiki it's called car locator, link will take you there, you can see how it's done, built app is in the bin folder.
Background is made with opengl shader.
Actually you can just click on my nickname and you'll see it.

**Comment by adywizard** [comment]
Just tested, on desktop the behavior is the same.

**Comment by podraco** [comment]
i'm currently working on the tap per view class, as it's broken rn.

i'm not sure if there's anyone working on the clock class. if you already have an idea on how to solve this, feel free to fork the repo and share the bug fix.

Otherwise, I would recommend you to wait until one our team members can take a look at the issue.

**Comment by adywizard** [comment]
I have some fix, it's not the best one but it's working. I'll post it later when I have some time as I don't have any pc with me.

**Comment by adywizard** [comment]
Hi there, if you want I added fix for this bug not sure if this is OK but it works.
https://github.com/adywizard/car-locator/blob/e4e9f38f85d61f2250b070f0b493438cbe6e8488/modified_picker/picker.py#L2409
it starts from this line also regex are changed, I used different way because of the Kivy's TextInput bug, view-port doesn't behave nice, so that would be my temporary solution.
video here:  https://www.youtube.com/watch?v=RkExWU8w0gc
let me know.
